Remember #107? well heres that again with less forced cmake hacks

yellows111 opened the pull request:
please comment on the gitignore situation with cmake?

yellows111 committed:
for ee7fbc93ae37672bf50e27cb0d309f9f8797a3e7

yellows111 committed:
let me correct that

AWA and SWA are currently still AL2

yellows111 committed:
Update .gitignore

yellows111 committed:
manpage update

yellows111 committed:
what

yellows111 committed:
seriously cmake? MORE files?

did i also mention that without manually finding each and every one you can't clean this up?

yellows111 committed:
more cmake caveats that are minor

cmake won't generate a cache in PWD unless told to do such.

yellows111 committed:
i've tested it, it works

*don't* be the PR-killer.......

Also, can't we remove the first arg from copy_dlls? they've not needed it for a long time.

modeco80 commented on src/main/CMakeLists.txt:
```diff
@@ -18,7 +18,16 @@ set_target_properties(collab-vm-server PROPERTIES
+
+# Dynamic binaries are SO much fun!
+if (WIN32)
+add_custom_command(TARGET collab-vm-server POST_BUILD
```
If you're REALLY gonna do this, 
use https://cmake.org/cmake/help/latest/module/GetPrerequisites.html

This is fragile, only works on MinGW, depends on bash, and depends on scripts I just nuked (because they deserved it.)
Wherehas GetPrerequisites works everywhere (although Windows is the only place we need to copy dlls because windows has to be special)

For vcpkg you'll have to maybe nuke this if that's detected (the Q&D way I will not merge because I'm sure there's some way that's infinitely better would probably involve regex testing CMAKE_TOOLCHAIN_FILE for `/scripts/buildsystems/vcpkg.cmake`) though because vcpkg will copy shared libraries automatically (on any platform). On linux/mac it may fuck up RPATH too so you can just kinda run it there but idk

yellows111 replied:
yeaaaaaaaaaah i had a feeling this would of been the thing i would of used and would of been bombed for.

using add_custom_command instead of looking into cmake's (what now) more than a thousand functions?

I mean cmake is flexible but it feels too flexible sometimes.

honestly i should of looked around more to see what cmake has support for and now that i know *it exists* i may make a copy_dlls cmake equiv if *possible*. Which it *should* be.

yellows111 replied:
>use https://cmake.org/cmake/help/latest/module/GetPrerequisites.html

```
CMake Error at src/main/CMakeLists.txt:29 (get_prerequisites):
  Unknown CMake command "get_prerequisites".
```

> Deprecated since version 3.16: Use file(GET_RUNTIME_DEPENDENCIES) instead.

Deprecated? I think theres an (undocumented) "removed"  missing from this.

also 

>file(GET_RUNTIME_DEPENDENCIES)

yeah i'm not following that deathtrap. (cannot tell difference between 32-bit and 64-bit DLLs, makes it suck. hard.)

yellows111 replied:
the alternative to most of this (except add_custom_command) is `$<TARGET_RUNTIME_DLLS:tgt>`
which has one sliiiiiiiight issue with it
> *New in version 3.21.*

also another problem with it:
is it meant to work? for me with or without list expansion it's empty

yellows111 replied:
Apparently I forgot to check if the "deathtrap" was revised, and as of writing, it was... in 3.21.
I should of probably mentioned this.

modeco80 commented on README.md:
```diff
@@ -8,7 +8,7 @@ CollabVM was coded and written by Cosmic Sans, Dartz, Geodude, and modeco80... A
 CollabVM Server is licensed under the [GNU Public License 3](https://www.gnu.org/licenses/gpl-3.0.en.html).
 
-The CollabVM Server Plugin Interface "CVMpi" is licensed under the [Lesser GNU Public License 3](https://www.gnu.org/licenses/lgpl-3.0.en.html).
+The CollabVM Server Plugin Interface "CVMpi" is licensed under the [Lesser GNU Public License 3](https://www.gnu.org/licenses/lgpl-3.0.en.html). 
```
I would reword this because the plugin interface isn't like. completely seperate from the server? the server source comes with and should optionally install sample plugin sources, something sorta like:

```diff
+ CollabVM Server is dual-licensed under the [GNU Public License 3](https://www.gnu.org/licenses/gpl-3.0.en.html) for the server, and the [LGPLv3](https://www.gnu.org/licenses/lgpl-3.0.en.html) license for the plugin interface headers, provided runtime code, and sample plugin code.
```

yellows111 replied:
i'll look into that.

yellows111 replied:
done https://github.com/yellows111/collab-vm-server/commit/b066e1632614be9ff484e36ce16d10cf9bfc4aee

yellows111 committed:
Merge branch 'master' of https://github.com/computernewb/collab-vm-server into master300-2

yellows111 committed:
i'm removing it

all alternatives to copy_dlls are a *LOSS*, they either:
1: don't exist anymore
2: are the WORST THING
3: flatout don't work as intended

yellows111 committed:
Merge pull request #13 from computernewb/master

webmerge upstream master to master300-2

yellows111 committed:
take modeco's suggestion for licensing segment

yellows111 committed:
thought this would be useful?

vars for versioning and add the equiv to "3.0.0-0" to current version?

yellows111 committed:
expand WIN32 to WIN32|MSYS|MINGW

minimal cross-compiling support?

modeco80 commented on src/main/CMakeLists.txt:
```diff
@@ -8,7 +8,7 @@ add_executable(collab-vm-server
 target_link_libraries(collab-vm-server PRIVATE ${CMAKE_THREAD_LIBS_INIT} Boost::program_options Boost::filesystem ${CMAKE_DL_LIBS} spdlog::spdlog fmt::fmt collabvm::core collabvm::websocket)
 
-if(WIN32)
+if(WIN32 OR MSYS OR MINGW)
```
I mean mingw enviroments would just shortcircuit when WIN32 is true anyways

yellows111 replied:
makes no sense in that context but apparently some cross-compiling envs specify either MSYS or MINGW and *not* WIN32 for some reason.

modeco80 commented on CMakeLists.txt:
```diff
@@ -9,7 +9,14 @@ endif()
+set(COLLABVM_VERSION_MAJOR 3)
+set(COLLABVM_VERSION_MINOR 0)
+set(COLLABVM_VERSION_PATCH 0)
+set(COLLABVM_PINT_VERSION 1)
```
Meh, the plugin ABI version is controlled in a .h file and I dunno if I wanna expose that anyways
We will only bump the abi (it's v0 right now) if a huge change is made either way

Also I'm going to be using git tags for versioning probably

yellows111 replied:
I just wanted to jump on (project VERSION), but we can use them around the cmake files in other ways than just (project VERSION).
